src/test/java/seedu/address/logic/commands/GenerateLocationsCommandTest.java

```diff
@@ -139,4 +151,53 @@ public void execute_invalidInputFromViewingAllQuarantined_throwCommandException(
         GenerateLocationsCommand command = new GenerateLocationsCommand(new IndexIdPairStub(INDEX_FIRST, null));
         assertThrows(CommandException.class, () -> command.execute(modelForAllQuarantined));
     }
+
+    @Test
+    public void execute_dateOfVisitIsWithinRange_success() {
+        String expectedMessage = "Generated locations for: Daniel Meier";
+        LocalDate testDate = LocalDate.now().minusDays(7);
+        Visit testVisitOne = new Visit(DANIEL, DANIEL_LOCATION, testDate);
+        model.addVisit(testVisitOne);
+        Model expectedModelForGenerate = expectedModel;
+        expectedModelForGenerate.addVisit(testVisitOne);
+        Predicate<Location> locationPredicate = location -> location.getId().equals(new Id("L456D"));
+        expectedModelForGenerate.updateFilteredLocationList(locationPredicate);
+        Index index = Index.fromOneBased(4);
+        GenerateLocationsCommand command = new GenerateLocationsCommand(new IndexIdPairStub(index, null));
+        CommandResult expectedCommandResult = new CommandResult(expectedMessage, false, false,
+                CommandResult.SWITCH_TO_VIEW_LOCATIONS);
+        assertCommandSuccess(command, model, expectedCommandResult, expectedModelForGenerate);
+    }
+
+    @Test
+    public void execute_dateOfVisitMoreThanTwoWeeksAgo_noVisitsFound() {
+        String expectedMessage = MESSAGE_PERSON_HAS_NO_VISITS;
+        LocalDate testDate = LocalDate.now().minusDays(14);
+        Visit testVisitOne = new Visit(DANIEL, DANIEL_LOCATION, testDate);
+        model.addVisit(testVisitOne);
+        Index index = Index.fromOneBased(4);
+        GenerateLocationsCommand command = new GenerateLocationsCommand(new IndexIdPairStub(index, null));
+        assertThrows(CommandException.class, () -> command.execute(model));
+        try {
+            command.execute(model);
+        } catch (CommandException e) {
+            assertTrue(e.getMessage().equals(expectedMessage));
+        }
```

Comment:
asserThrows method from commandutils can be used here.

Reply:
There are also other code locations that can use assertThrows

Reply:
The assertThrows method is overloaded to also take in the expected message of the exception thrown. Using this can help combine your previous assertThrow method and the try catch block into one line.

Reply:
The assertThrows method is overloaded to also take in the expected message of the exception thrown. Using this can help combine your previous assertThrow method and the try catch block into one line.

Reply:
Done!